manifests/cirunner.pp

```diff
@@ -34,52 +34,36 @@
 # Copyright 2015 Tobias Brunner, VSHN AG
 #
 class gitlab::cirunner (
-  $concurrent = undef,
-  $metrics_server = undef,
-  $hiera_default_config_key = 'gitlab_ci_runners_defaults',
-  $hiera_runners_key = 'gitlab_ci_runners',
-  $manage_docker = true,
-  $manage_repo = true,
-  $xz_package_name = 'xz-utils',
-  $package_ensure = installed,
-  $package_name = 'gitlab-runner',
+  Optional[Integer]       $concurrent           = undef,
+  Optional[String]        $metrics_server       = undef,
+  Boolean                 $manage_repo          = true,
+  String                  $conf_file            = '/etc/gitlab-runner/config.toml',
+  Enum[installed, absent] $package_ensure       = installed,
+  String                  $package_name         = 'gitlab-runner',
+  Optional[String]        $default_token        = undef,
+  String                  $default_url          = 'https://gitlab.com',
+  Boolean                 $default_run_untagged = true,
+  Boolean                 $default_locked       = false,
+  Optional[Array[String]] $default_tags         = undef,
+  Hash                    $docker_runners       = {},
+  Hash                    $shell_runners        = {},
+  Hash                    $ssh_runners          = {},
+  Hash                    $docker_ssh_runners   = {},
+  Hash                    $parallels_runners    = {},
+  Hash                    $virtualbox_runners   = {},
+  Hash                    $kubernetes_runners   = {},
 ) {
 
-  validate_string($hiera_default_config_key)
```

Comment:
removing those validate calls should be made with the datatypes for old params in another PR.

Reply:
cirunner.pp is a complete re-write so im not sure a dedicated PR makes sense